src/erc721_utils.rs

```diff
@@ -0,0 +1,298 @@
+//! This module contains utility functions for interacting with ERC721 tokens and contracts
+use crate::jsonrpc::error::Web3Error;
+use crate::{client::Web3, types::SendTxOption};
+use clarity::{abi::encode_call, PrivateKey as EthPrivateKey};
+use clarity::{Address, Uint256, abi::Token};
+use std::time::Duration;
+use tokio::time::timeout as future_timeout;
+use clarity::Address as EthAddress;
+
+pub static ERC721_GAS_LIMIT: u128 = 100_000;
```

Comment:
this whole constant should probably just be replace with gas estimation. But we'll keep it for now. This is more of a comment to myself.